addons/sale_timesheet/views/project_task_views.xml

```diff
@@ -88,22 +88,28 @@
             <field name="inherit_id" ref="hr_timesheet.view_task_form2_inherited"/>
             <field name="arch" type="xml">
                 <xpath expr="//field[@name='child_ids']/tree//field[@name='remaining_hours']" position="after">
-                    <field name="remaining_hours_so" widget="timesheet_uom" optional="hide" groups="base.group_user"/>
+                    <field name="sale_line_id" invisible="1"/>
+                    <field name="remaining_hours_available" invisible="1"/>
+                    <field name="remaining_hours_so" widget="timesheet_uom" optional="hide" groups="base.group_user"  attrs="{'invisible': ['|', ('sale_line_id', '=', False), ('remaining_hours_available', '=', False)]}"/>
                 </xpath>
                 <xpath expr="//field[@name='depend_on_ids']/tree//field[@name='remaining_hours']" position="after">
-                    <field name="remaining_hours_so" widget="timesheet_uom" optional="hide" groups="base.group_user"/>
+                    <field name="sale_line_id" invisible="1"/>
+                    <field name="remaining_hours_available" invisible="1"/>
+                    <field name="remaining_hours_so" widget="timesheet_uom" optional="hide" groups="base.group_user" attrs="{'invisible': ['|', ('sale_line_id', '=', False), ('remaining_hours_available', '=', False)]}"/>
                 </xpath>
 
             </field>
         </record>
 
-        <record id="view_task_tree2_inherited" model="ir.ui.view">
+        <record id="project_task_view_tree_base_inherit" model="ir.ui.view">
```

Comment:
Changing the view name would require an upgrade script, I'm not sure if it's worth it